controllers/urls.js

```diff
@@ -0,0 +1,131 @@
+const { Url, User } = require("../models/schema");
+const shuffle = require("../lib/utils").shuffle;
+const TagControl = require("./tags");
+const UserControl = require("./users");
+
+const createURL = async (req, res) => {
+  try {
+    const [title, url, tags] = req.body.params;
+    const submittedBy = req.body.userId;
+    const existingUrl = await Url.findOne({ url });
+    if (existingUrl) {
+      return res.status(400).json({ error: "URL already exists" });
+    }
+
+    const newUrl = await Url.create({ title, url, submittedBy, tags });
+
+    TagControl.attachURL(tags, newUrl.id);
+    UserControl.attachURL(submittedBy, newUrl.id);
+
+    return res.status(201).json(newUrl);
+  } catch (err) {
+    console.error(err);
+    return res.status(500).json({ error: "Server error" });
+  }
+};
+
+const deleteURL = async (req, res) => {
+  try {
+    const { urlId } = req.params;
+    const url = await Url.findById(urlId);
+    if (!url) {
+      return res.status(404).json({ error: "URL not found" });
+    }
+
+    await TagControl.detachURL(url.tags, urlId);
+    await UserControl.detachURL(url.submittedBy, urlId);
+
+    await Url.deleteOne({ _id: urlId });
+
+    return res.status(200).json({ message: "URL removed successfully" });
+  } catch (err) {
+    console.error(err);
+    return res.status(500).json({ error: "Server error" });
+  }
+};
+
+const __getURLsFromDb = async (tags, page = 1, limit = 100) => {
+  if (!tags) tags = "all";
```

Comment:
I checked that when no tags are selected frontend basically  sends all tags in request so kinda like this ```localhost:8000/api/mix?tag=tag1&tag=tag2..... so on``` .  So do we need "all" ? since it will always execute url.find({tags: {$in: tags} ) instaed of Url.find()